### ISSUE TITLE
Big overfit on new/future data

### ISSUE THREAD

**Issue by processadd** [jdb78/pytorch-forecasting#903]
PyTorch-Forecasting version: current master branch
PyTorch version: 1.9.0
Python version: 3.8

### Expected behavior

Trained model can predict future/new data with acceptable loss.

### Actual behavior

We have some hourly data and want to predict single target `T`. These hourly data are summary values for some entities and hence no features can identify an identity. In fact, I specified a single column `G` as group_ids. The target value range is around `6000~9000`. I use `24*14` to predict `24*3`. The findings are:
1. If using 2 year data, the loss is hard to decrease at `~100` (lr is `3E-05`). 
2. If using 1 year data, the loss can decrease to `23` with `epochs=700`.  
3. After stopped at `700` epoch and got loss between 20 and 30, like in TensorBoard
<img width="414" alt="image" src="https://user-images.githubusercontent.com/27667124/159772163-fd490213-058f-4ff4-b384-eb822f27f775.png">

apply the model to trained data shows some ideal result like
<img width="1101" alt="image" src="https://user-images.githubusercontent.com/27667124/159769637-bf3a76bc-6fad-4d21-882c-b0ba72d90f21.png">
or
<img width="1019" alt="image" src="https://user-images.githubusercontent.com/27667124/159769587-96f1bb99-9a38-4b3f-bebf-a1a9073b5b4b.png">

For new data close to the train data, e.g. train data ends at 10/31, if predict `24*3` hours from 11/1, it is generally ok except for few hours with relatively big loss, but for "remote" future, e.g. use the trained model to predict data from 12/1, the predict is not acceptable at all like
<img width="741" alt="image" src="https://user-images.githubusercontent.com/27667124/159770621-e0714d42-3d12-401f-83e8-7fe1d2265bef.png">

Any suggestions for possible improvement or the data just cannot get trained?

Thanks

**Comment by processadd** [comment]
After debugging found this should be related with some test data error. In training data we wrongly use avg_by_day is on `dt.day`, on testing data this definitely not work.
